src/Microsoft.Health.Dicom.SqlServer/Registration/DicomSqlServerRegistrationExtensions.cs

```diff
@@ -43,17 +43,27 @@ public static class DicomSqlServerRegistrationExtensions
         IServiceCollection services = EnsureArg.IsNotNull(dicomServerBuilder, nameof(dicomServerBuilder)).Services;
 
         // Add core SQL services
+        IConfigurationSection config = EnsureArg
+            .IsNotNull(configurationRoot, nameof(configurationRoot))
+            .GetSection(SqlServerDataStoreConfiguration.SectionName);
+
         services
             .AddSqlServerConnection(
-                config =>
+                sqlOptions =>
                 {
-                    configurationRoot?.GetSection(SqlServerDataStoreConfiguration.SectionName).Bind(config);
-                    configureAction?.Invoke(config);
+                    config.Bind(sqlOptions);
+                    configureAction?.Invoke(sqlOptions);
                 })
             .AddSqlServerManagement<SchemaVersion>()
             .AddSqlServerApi()
             .AddBackgroundSqlSchemaVersionResolver();
 
+        // Optionally enable workload identity
+        DicomSqlServerOptions options = new();
+        config.Bind(options);
+        if (options.EnableWorkloadIdentity)
+            services.EnableWorkloadManagedIdentity();
```

Comment:
Instead of looking at SQL specifically, I think it's better if we look at the service's configuring authentication scheme. Although as I type this, karma will dictate there will be some incident in which I need to verify this for SQL. However, I could always look at the SQL client event souce.